### PR TITLE
Fix Keras scheduler import so it works for older versions of Keras

### PR DESCRIPTION
**Pull request by Rocketknight1**
See bug report [here](https://github.com/huggingface/transformers/pull/28588#issuecomment-1924842630)

This one caught me out - we used to import from `tf.keras.optimizers.schedules`, which worked on older versions of Keras. However, as part of the Keras 3 compatibility patch, we moved all `tf.keras` imports to `keras` imports, which normally should make no difference. However, in older versions of Keras the schedule classes actually live in `keras.optimizers.schedules.learning_rate_schedule` (even though they're in the right place in the equivalent version of `tf.keras`). I had no idea about this, and didn't catch it in my testing because I was using newer versions of TF!

Thanks to @echan5 for the warning!

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
🧼 thanks
